src/dolphin/workflows/config.py

```diff
@@ -0,0 +1,524 @@
+        "%Y%m%d",
+        description="Format of dates contained in CSLC filenames",
+    )
+
+    mask_files: List[str] = Field(
+        default_factory=list,
+        description=(
+            "List of mask files to use, where convention is"
+            " 0 for no data/invalid, and 1 for data."
+        ),
+    )
+
+    class Config:
+        extra = Extra.forbid  # raise error if extra fields passed in
+
+    # validators
+    @validator("cslc_file_list", pre=True)
+    def _check_input_file_list(cls, v):
+        if v is None:
+            return []
+        if isinstance(v, (str, Path)):
+            v_path = Path(v)
+            # Check if it's a newline-delimited list of input files
+            if v_path.exists() and v_path.is_file():
+                filenames = [Path(f) for f in v_path.read_text().splitlines()]
+                # If given as relative paths, make them relative to the file
+                parent = v_path.parent
+                return [parent / f if not f.is_absolute() else f for f in filenames]
+            else:
+                raise ValueError(
+                    f"Input file list {v_path} does not exist or is not a file."
+                )
+
+        return [Path(f) for f in v]
+
+    @validator("subdataset", pre=True, always=True)
+    def _check_for_opera(cls, v, values):
+        cslc_file_list = values.get("cslc_file_list")
+        # if we're not dealing with all OPERA files, just return whatever they gave
+        if any(re.search(OPERA_BURST_RE, str(f)) is None for f in cslc_file_list):
+            return v
+        # Here we're dealing with all OPERA files, so we need to set the subdataset
+        if v is None:
+            # Assume that the user forgot to set the subdataset, and set it to the
+            # default OPERA dataset name
+            logger.info(
+                "CSLC files look like OPERA files, setting subdataset to"
+                f" {OPERA_DATASET_NAME}."
+            )
+            return OPERA_DATASET_NAME
+        return v
+
+    @validator("mask_files", pre=True)
+    def _check_mask_files(cls, v):
+        if isinstance(v, (str, Path)):
+            # If they have passed a single mask file, return it as a list
+            return [Path(v)]
+        elif v is None:
+            return []
+        return [Path(f) for f in v]
+
+    @root_validator
+    def _check_slc_files_exist(cls, values):
+        file_list = values.get("cslc_file_list")
+        date_fmt = values.get("cslc_date_fmt")
+        if not file_list:
+            raise ValueError("Must specify list of input SLC files.")
+
+        # Filter out files that don't have dates in the filename
+        file_matching_date = [Path(f) for f in file_list if get_dates(f, fmt=date_fmt)]
+        if len(file_matching_date) < len(file_list):
+            raise ValueError(
+                f"Found {len(file_matching_date)} files with dates in the filename"
+                f" out of {len(file_list)} files."
```

Comment:
Why do the input files need to have some date string in their filename? Why do we care about the filenames at all?

EDIT: Okay so we want dates in the filenames in order to sort the files in chronological order. But why not just require users to specify the files in sorted order? That seems like the simplest, most straightforward approach. Doing it this way just seems to make the interface slightly less specific and the implementation slightly more error-prone.